src/Raven.Server/ServerWide/Commands/ConnectionStrings/PutConnectionStringCommand.cs

```diff
@@ -146,4 +149,53 @@ public override void UpdateDatabaseRecord(DatabaseRecord record, long etag)
             record.SnowflakeConnectionStrings[ConnectionString.Name] = ConnectionString;
         }
     }
+
+    public sealed class PutAiConnectionStringCommand : PutConnectionStringCommand<AiConnectionString>
+    {
+        public PutAiConnectionStringCommand()
+        {
+            // for deserialization
+        }
+
+        public PutAiConnectionStringCommand(AiConnectionString connectionString, string databaseName, string uniqueRequestId) : base(connectionString, databaseName, uniqueRequestId)
+        {
+
+        }
+
+        public override void UpdateDatabaseRecord(DatabaseRecord record, long etag)
+        {
+            var isUpdate = record.AiConnectionStrings.TryGetValue(ConnectionString.Name, out var oldAiConnectionString) && oldAiConnectionString != null;
```

Comment:
shouldn't all of the validation be in the endpoint? @ArieSLV check with Karmel

Reply:
Done.